src/AnyTVKue.js

```diff
@@ -65,6 +65,50 @@ class AnyTVKue {
 
         this.queue.create.isCustom = true;
 
+        /**
+         * Patch created job's backoff function to support custom backoff types
+         */
+
+        let _createJob = this.queue.createJob;
+
+        this.queue.createJob = function (name, payload) {
+
+            let job = _createJob.call(this, name, payload);
+            let _backoff = job.backoff;
+
+            job.backoff = function (backoff_type) {
+
+                let backoff;
+                switch (backoff_type) {
+
+                    /**
+                     * Custom callback functions
+                     * Executed by eval() so cannot contain variables i.e. can't use config
+                     * https://github.com/Automattic/kue#failure-backoff
+                     */
+
+                    // 2 mins, 4 mins, 8m mins...
+                    case 'fixed_doubling':
+                        backoff = attempts => Math.pow(2, attempts) * 1000 * 60;
+                        break;
+
+                    // custom exponential based on initial delay
+                    // starts at 2 seconds when there's no initial delay
+                    case 'delay_doubling':
+                        backoff = (attempts, delay) => delay ? delay * 2 : 2000; // milliseconds
```

Comment:
The 2s default should be in config

Reply:
Not possible with kue, unfortunately

https://github.com/anyTV/anytv-node-kue/blob/2b9e138fd18eef7c1d8a28b53d6512926bdd763d/src/AnyTVKue.js#L84-L88

Reply:
I'm pretty sure we can read it when we stub the createJob function?

Reply:
Also, don't we also need to evaluate attempts as well?

Reply:
Noted, will look into it.

Reply:
eval with passed in scope
https://stackoverflow.com/a/25859853

Reply:
Update:
Tried different approaches but I think I'm out of my depth for this one.
Perhaps a different ticket?

The eval with passed in scope answers all create a wrapper for eval that accepts context, however kue just uses eval() as is.

The change would probably need to be done on kue itself, specifically this file:

[/lib/queue/job.js](https://github.com/Automattic/kue/blob/87d61503d3d9cc024633efc7611bd25551f0f87d/lib/queue/job.js#L207-L211)
```js
...
      if( hash.backoff ) {
        var source = 'job._backoff = ' + hash.backoff + ';';
//                require('vm').runInContext( source );
        eval(source);
      }
...
```

Reply:
After looking at their code, the way they check is that if backoff is a isPlainObject, otherwise they call toString
https://github.com/Automattic/kue/blob/master/lib/queue/job.js#L854

So effectively we can just build our own string first, which they would just use directly

ex:

```
let backoff_string = '(attempts, delay) => delay ? delay * 2 : ' + config.defaultTime + ';';
```

Then it'll just be pass to eval() directly

Reply:
Awesome, it works, thanks. Updated 82d82ef